autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Protocol
+
+from autogen.agentchat import ConversableAgent
+
+from .graph_query_engine import GraphQueryEngine
+
+
+class GraphRagAgent(ConversableAgent, Protocol):
+    """
+    A graph rag agent is a conversable agent which could query graph database for answers.
+
+    An implementing agent class would
+    1. create a graph in the underlying database with input documents
+    2. use the retrieve() method to retrieve information.
+    3. use the retrieved information to generate and send back messages.
+
+    For example,
+    graph_query_engine = GraphQueryEngine(...)
+    graph_query_engine.init_db([Document(doc1), Document(doc2), ...])
+
+    graph_rag_agent = GraphRagAgent(
+        name="graph_rag_agent",
+        max_consecutive_auto_reply=3,
+        ...
+    )
+    graph_rag_agent.attach_graph_query_engine(graph_query_engine)
```

Comment:
Is it possible to achieve the same functionality with the Capability design? It's a more generic and composable way to extend agent's capabilities.